docs/content/en/docs-dev/user-guide/managing-application/customizing-deployment/script-run.md

```diff
@@ -55,6 +55,52 @@ The commands run in the directory where this application configuration file exis
 
 ![](/images/script-run.png)
 
+### Execute the script file
+
+If your script is so long, you can separate the script as a file.
+You can put the file with the app.pipecd.yaml in the same dir and then you can execute the script like this.
+
+```
+apiVersion: pipecd.dev/v1beta1
+kind: KubernetesApp
+spec:
+  name: script-run
+  pipeline:
+    stages:
+      - name: SCRIPT_RUN
+        with:
+          run: |
+            sh script.sh
+```
+
+```
+.
+├── app.pipecd.yaml
+└── script.sh
+```
+
+## Builtin commands
+
+Currently, you can use the commands which are installed in the environment for the piped.
+
+For example, If you are using the container platform and the offcial piped container image, you can use the command below.
+- git
+- ssh
+- jq
+- curl
+- kubectl, helm, kustomize, terraform stored in $PIPED_TOOL_DIR 
+  - Be careful. These tools are sometimes absent because they are installed when the first deployment has been done using each tool. Please check $PIPED_TOOL_DIR before you use them.
+  - Binaries can be specified as "kubectl-${version}" or "kubectl". 
+  - If no version is specified on the piped config or app.pipecd.yaml, both the ones with and without default version suffix will be installed. 
+  - Please check the [code](https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/toolregistry/install.go#L28C1-L33C2) about the default versions.
+- and the builtin commands installed in the base image.
```

Comment:
```suggestion
- git
- ssh
- jq
- curl
- commands installed by piped in $PIPED_TOOL_DIR (check at runtime)
- built-in commands installed in the base image.
```

How about this? @ffjlabo

Reply:
@khanhtc1202 
I think we also have to mention that the tools can be absent sometimes.
WDYT?

Reply:
That is why I added the note `(check at runtime)`. I think it would be better not to confuse users by showing too much explanation 🤔

Reply:
OK, I agree with your point.
I will fix it👌

Reply:
fixed [48022da](https://github.com/pipe-cd/pipecd/pull/5127/commits/48022daafc84b0beb3e27cb973d4f9e6979a742d)